stmhal/can.c

```diff
@@ -525,6 +556,52 @@ STATIC mp_obj_t pyb_can_setfilter(mp_uint_t n_args, const mp_obj_t *pos_args, mp
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_KW(pyb_can_setfilter_obj, 1, pyb_can_setfilter);
 
+STATIC mp_obj_t pyb_can_rxcallback(mp_obj_t self_in, mp_obj_t fifo_in, mp_obj_t callback_in) {
+    pyb_can_obj_t *self = self_in;
+    mp_int_t fifo = mp_obj_get_int(fifo_in);
+
+
+    if (callback_in == mp_const_none){
+        if (fifo == 0){
+            __HAL_CAN_DISABLE_IT(&self->can, CAN_IT_FMP0);
+            self->rxcallback0 = mp_const_none;
+        } else if (fifo == 1){
+            __HAL_CAN_DISABLE_IT(&self->can, CAN_IT_FMP1);
+            self->rxcallback1 = mp_const_none;
+        }
+    } else if (mp_obj_is_callable(callback_in)) {
+        if (self->can_id == PYB_CAN_1){
+            if (fifo == 0) {
+                self->rxcallback0 = callback_in;
```

Comment:
> I guess it is a matter of style. Personally I think this way of writing code is a bit easier to read an understand, but I will not make any noise if you want me to change it before accepting the code.

Well, I'd say that writing code such way hides the fact that what those if's do is actually transcoding values, but operations on ready values are the same. Suppose you'd need to change these operations (add another one, or rename a function) - these would need to be done in 4 places (what if there were more?). In more complex cases someone else doing a refactor could even miss to update a branch. Finally, doing multiple calls will almost certainly costs more bytes - and it seems that to completely wrap CAN there will be need quite a few of them, and there're lot of other hardware to wrap yet...

Anyway, it's up to @dpgeorge to consider how that's important.

Anyway, I started to look at your patch because for esp8266 port there soon would be need for callbacks, and that's where there're lot of boilerplate code and what will need to be refactored on uPy level...